apps/dcos_overlay/src/dcos_overlay_poller.erl

```diff
@@ -171,20 +171,76 @@ maybe_create_vtep(_Overlay = #mesos_state_agentoverlayinfo{backend = Backend},
     VTEPNameStr = binary_to_list(VTEPName),
     ParsedVTEPMAC = list_to_tuple(parse_vtep_mac(VTEPMAC)),
     {ParsedVTEPIP, PrefixLen} = parse_subnet(VTEPIP),
-    dcos_overlay_netlink:iplink_add(Pid, VTEPNameStr, "vxlan", VNI, 64000),
-    {ok, _} = dcos_overlay_netlink:iplink_set(Pid, ParsedVTEPMAC, VTEPNameStr),
-    {ok, _} = dcos_overlay_netlink:ipaddr_replace(Pid, inet, ParsedVTEPIP, PrefixLen, VTEPNameStr),
-    case {application:get_env(dcos_overlay, enable_ipv6, true), VTEPIP6} of
+    case vxlan_interface_present(Pid, VXLan) of
+        false ->
+            dcos_overlay_netlink:iplink_add(Pid, VTEPNameStr, "vxlan", VNI, 64000),
+            {ok, _} = dcos_overlay_netlink:iplink_set(Pid, ParsedVTEPMAC, VTEPNameStr);
+        {true, match} ->
+            lager:debug("Attributes of ~p is up-to-date.", [VTEPNameStr]);
+        {true, notmatch} ->
+            lager:debug("Attributes of ~p are not update-to-date, and will be recreated.", [VTEPNameStr]),
+            dcos_overlay_netlink:iplink_delete(Pid, VTEPNameStr),
```

Comment:
Removing an existing vtep1024 interface might break customer setting if the customer is using this interface for some other purpose and happens to install DC/OS. We would like to fail explicitly instead of silently changing customer setting. 

We should just check if vtep1024 is already present and configure only if it is not present.

Reply:
If we have a recognizable name for vtep, e.g. dcos.1024, we won't be afraid of removing consumer-defined interfaces. And all these code works fun, and should remain.

Reply:
@GoelDeepak is it still the case?